frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseRepositoryAction.java

```diff
@@ -0,0 +1,292 @@
+                .getAction(actionIdFormatString.format(PushCurrentAction.class.getSimpleName()));
+            pushAction.actionPerformed(anActionEvent);
+            break;
+          default :
+            break;
+        }
+
+        break;
+
+      case BehindRemote :
+        boolean pullApproved = true;
+        if (PropertiesComponent.getInstance().getBoolean(SHOW_PULL_APPROVAL, /* defaultValue */ true)) {
+          val pullApprovalDialogBuilder = MessageDialogBuilder.okCancel(
+              getString("action.GitMachete.TraverseRepositoryAction.dialog.pull-verification.title"),
+              getString("action.GitMachete.TraverseRepositoryAction.dialog.pull-verification.text.HTML")
+                  .format(gitMacheteBranch.getName(), remoteTrackingBranch.getRemoteName(), remoteTrackingBranch.getName()));
+
+          pullApprovalDialogBuilder
+              .yesText(getString("action.GitMachete.TraverseRepositoryAction.dialog.pull-verification.ok-text"))
+              .icon(Messages.getQuestionIcon()).doNotAsk(new PullApprovalDialog());
+
+          pullApproved = pullApprovalDialogBuilder.ask(getProject(anActionEvent));
+        }
+
+        if (pullApproved) {
+          val pullAction = ActionManager.getInstance()
+              .getAction(actionIdFormatString.format(PullCurrentAction.class.getSimpleName()));
+          pullAction.actionPerformed(anActionEvent);
+        }
+        break;
+      default :
+        break;
+    }
+    refreshBranchLayout(anActionEvent);
+  }
+
+  @UIEffect
+  private void syncBranchToParent(@ConfirmedNonRoot IManagedBranchSnapshot gitMacheteBranch, AnActionEvent anActionEvent) {
+    val gitMacheteNonRootBranch = gitMacheteBranch.asNonRoot();
+    val syncToParentStatus = gitMacheteNonRootBranch.getSyncToParentStatus();
+    switch (syncToParentStatus) {
+      case MergedToParent :
+        val slideOutCurrentAction = ActionManager.getInstance()
+            .getAction(actionIdFormatString.format(SlideOutCurrentAction.class.getSimpleName()));
+        slideOutCurrentAction.actionPerformed(anActionEvent);
+        break;
+      case InSyncButForkPointOff :
+        val overrideForkPointAction = ActionManager.getInstance()
+            .getAction(actionIdFormatString.format(OverrideForkPointOfCurrentAction.class.getSimpleName()));
+        overrideForkPointAction.actionPerformed(anActionEvent);
+        break;
+      case OutOfSync :
+        val nonRootBranch = gitMacheteBranch.asNonRoot();
+
+        val selectedAction = new DivergedFromParentDialog(getProject(anActionEvent), nonRootBranch.getParent(), nonRootBranch)
+            .showAndGetThePreferredAction();
+        if (selectedAction == null) {
+          log().debug(
+              "Action selected for resolving divergence from parent is null: most likely the action has been canceled from Diverge-From-Remote-Dialog dialog");
+          return;
+        }
+        switch (selectedAction) {
+          case RESET_TO_PARENT :
```

Comment:
Hmmm this is rather unlikely to what user ever wants... giving this option even have actively **negative** value, if users decide to use this action, they're likely to "lose" commits. I mean, [stuff is rarely if ever truly lost in git](https://virtuslab.github.io/tips/#git/git-reflog), but still

Reply:
ok, no problem. I would remove it.